UnattendResources/Logon.ps1

```diff
@@ -74,47 +118,103 @@ function Release-IP {
         }
 }
 
-try
-{
-    Import-Module "$resourcesDir\ini.psm1"
-    $installUpdates = Get-IniFileValue -Path $configIniPath -Section "DEFAULT" -Key "InstallUpdates" -Default $false -AsBoolean
-    $persistDrivers = Get-IniFileValue -Path $configIniPath -Section "DEFAULT" -Key "PersistDriverInstall" -Default $true -AsBoolean
+function Install-WindowsUpdates {
+    Import-Module "$resourcesDir\WindowsUpdates\WindowsUpdates"
+    $BaseOSKernelVersion = [System.Environment]::OSVersion.Version
+    $OSKernelVersion = ($BaseOSKernelVersion.Major.ToString() + "." + $BaseOSKernelVersion.Minor.ToString())
 
-    if($installUpdates)
+    #Note (cgalan): We are adding some updates to a blacklist because
+    # we have observed that they will break the operation system.
+    $KBIdsBlacklist = @{
+        "6.3" = @("KB2887595")
+    }
+    $excludedUpdates = $KBIdsBlacklist[$OSKernelVersion]
+    $updates = ExecRetry {
+        Get-WindowsUpdate -Verbose -ExcludeKBId $excludedUpdates
+    } -maxRetryCount 30 -retryInterval 1
+    $maximumUpdates = 100
+    if (!$updates.Count) {
+        $updates = [array]$updates
+    }
+    if ($updates) {
+        $availableUpdatesNumber = $updates.Count
+        Write-Host "Found $availableUpdatesNumber updates. Installing..."
+        try {
+            #Note (cgalan): We have observed that in case the update fails to install
+            #we need to restart the computer in order to apply the eariler ones

```

Comment:
In case the update fails, we need to reboot the instance in order for the updates to be retrieved on a changed system state and be applied correctly.